engine/runtime-compiler/src/main/java/org/enso/compiler/dump/DocsGenerate.java

```diff
@@ -0,0 +1,107 @@
+package org.enso.compiler.dump;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.IdentityHashMap;
+import org.enso.compiler.context.CompilerContext;
+import org.enso.compiler.core.IR;
+import org.enso.compiler.core.ir.Module;
+import org.enso.compiler.core.ir.module.scope.Definition;
+import org.enso.compiler.core.ir.module.scope.definition.Method;
+import org.enso.filesystem.FileSystem;
+import org.enso.pkg.QualifiedName;
+import scala.collection.immutable.Seq;
+import scala.jdk.CollectionConverters;
+
+/** Generator of documentation for an Enso project. */
+public final class DocsGenerate {
+  private DocsGenerate() {}
+
+  /**
+   * Iterate over all provide modules and generate documentation using {@code pkg}'s {@link
+   * FileSystem}.
+   *
+   * @param <File> abstract file to operate with
+   * @param visitor visitor to use to generate the output
+   * @param pkg library to generate the documentation for
+   * @param modules parsed modules found in the library
+   * @return directory where the output was generated
+   * @throws IOException when I/O problem occurs
+   */
+  public static <File> File write(
+      DocsVisit visitor, org.enso.pkg.Package<File> pkg, Iterable<CompilerContext.Module> modules)
+      throws IOException {
+    var fs = pkg.fileSystem();
+    var docs = fs.getChild(pkg.root(), "docs");
+    var api = fs.getChild(docs, "api");
+    fs.createDirectories(api);
+
+    for (var module : modules) {
+      var ir = module.getIr();
+      assert ir != null : "need IR for " + module;
+      if (ir.isPrivate()) {
+        continue;
+      }
+      var moduleName = module.getName();
+      var md = fs.getChild(api, moduleName + ".md");
```

Comment:
Should we convert `.` in module name into `/` (subdirectories)? Then such docs would resemble the structure of source files.

Or is keeping them in a single directory preferred for some reason? Why?

Reply:
No, using single directory is not preferred for any reason. We can use `/` and subdirectories if that's the preference of the reviewers.

Reply:
Changed to subdirectories in 42a9464cbf